panels/info/cc-info-overview-panel.c

```diff
@@ -157,55 +158,140 @@ update_vendor_specific_info (CcInfoOverviewPanel *self)
+      g_warning ("%s", error_root->message);
+      return NULL;
+    }
+
+  hyphen_index = index (image_version, '-');
+  if (hyphen_index == NULL)
+    return NULL;
+
+  return g_strndup (image_version, hyphen_index - image_version);
+}
+
+static char *
+find_terms_document_for_language (const gchar *product_name,
+                                  const gchar *language)
+{
+  const gchar * const * data_dirs;
+  gchar *path = NULL;
+  g_autofree gchar *filename;
+  gint i;
+
+  data_dirs = g_get_system_data_dirs ();
+  filename = g_strconcat ("Terms-of-Use-", language, ".pdf", NULL);
+
+  for (i = 0; data_dirs[i] != NULL; i++)
+    {
+      path = g_build_filename (data_dirs[i],
+                               "eos-license-service",
+                               "terms",
+                               product_name,
+                               filename,
+                               NULL);
+
+      if (g_file_test (path, G_FILE_TEST_EXISTS))
+        return path;
+
+      g_free (path);
+    }
+
+  return NULL;
+}
+
+static char *
+find_terms_document_for_languages (const gchar *product_name,
+                                   const gchar * const *languages)
+{
+  int i;
+  gchar *path = NULL;
+
+  if (product_name == NULL)
+    return NULL;
+
+  for (i = 0; languages[i] != NULL; i++)
+    {
+      path = find_terms_document_for_language (product_name, languages[i]);
+
+      if (path != NULL)
+        return path;
+    }
+
+  return NULL;
+}
+
 static gboolean
 on_attribution_label_link (GtkLabel            *label,
                            gchar               *uri,
                            CcInfoOverviewPanel *self)
 {
   g_autoptr(GError) error = NULL;
   const gchar * const * languages;
-  const gchar * const * data_dirs;
-  const gchar *language;
   g_autofree gchar *pdf_uri = NULL;
   g_autofree gchar *path = NULL;
-  gint i, j;
-  gboolean found = FALSE;
+  g_autofree gchar *product_name = NULL;
 
   if (g_strcmp0 (uri, "attribution-link") != 0)
     return FALSE;
 
-  data_dirs = g_get_system_data_dirs ();
   languages = g_get_language_names ();
+  product_name = get_product_name ();
   path = NULL;
 
-  for (i = 0; languages[i] != NULL; i++)
-    {
-      language = languages[i];
-
-      for (j = 0; data_dirs[j] != NULL; j++)
-        {
-          path = g_build_filename (data_dirs[j],
-                                   "eos-license-service",
-                                   "terms",
-                                   language,
-                                   "Endless-Terms-of-Use.pdf",
-                                   NULL);
-
-          if (g_file_test (path, G_FILE_TEST_EXISTS))
-            {
-              found = TRUE;
-              break;
-            }
-
-          g_free (path);
-          path = NULL;
-        }
-
-      if (found)
-        break;
-    }
+  path = find_terms_document_for_languages (product_name, languages);
+  if (path == NULL && g_strcmp0 ("eos", product_name) != 0)
```

Comment:
See my comment in the other PR, I don't think the extra `g_strcmp0()` check is really needed.